### PR TITLE
feat: add rename-file command

### PR DESCRIPTION
**Pull request by CodeByZach**
Rename a file or directory and ensure Git recognizes the change, regardless of filesystem case-sensitivity. It combines the functionality of the `mv` command and `git mv`. This is particularly useful for renaming files or directories to change only their case, which might not be detected by Git on case-insensitive filesystems.

I initially considered naming it `git-mv` due to its intuitive functionality. However, I noticed that the naming convention in the repository includes both commands with and without the dash following "git-". To avoid conflict with the native `git mv` command, I opted for `git-rename-file` instead.

- Added new command 'rename-file' to rename files and directories within a Git repository regardless of filesystem case-sensitivity.
- Updated AUTHORS and documentation in Commands.md.
- Implemented bash, zsh, and fish completions for 'rename-file'.
- Created manual pages for 'rename-file' in various formats (man, html, md).
- Updated index.txt to include 'rename-file' command.

### CODE REVIEW

File: bin/git-rename-file
Comment (spacewander) [inline bin/git-rename-file:2]: Let's add https://github.com/tj/git-extras/blob/e1a4e7195d54cf3e0a931bb53a2584697ba431f0/bin/git-rename-remote#L3 for new command

File: bin/git-rename-file
Comment (spacewander) [inline bin/git-rename-file:51]: Could we use `find "$dir"` to save a child shell?

File: bin/git-rename-file
Comment (CodeByZach) [reply bin/git-rename-file:2]: I'm good with this.

File: bin/git-rename-file
Comment (CodeByZach) [reply bin/git-rename-file:51]: Can you show me what you mean? Thanks!

Comment (spacewander) [reply bin/git-rename-file:51]: We can rewrite the cmd into `find "$dir" -maxdepth 1 -name "$base" ...` so we can avoid using a sub shell wrapped with `()`

Comment (hyperupcall) [reply bin/git-rename-file:51]: Is `| grep -q $base` any different from redirecting like `>/dev/null 2>&1`? I believe `find` should already filter out the path;

File: bin/git-rename-file
Comment (hyperupcall) [inline bin/git-rename-file:94]: Once `set -e` is added to the top, these two `exit 1`'s can be removed;

File: bin/git-rename-file
Comment (hyperupcall) [inline bin/git-rename-file:32]:
```suggestion
    usage >&2
```

File: bin/git-rename-file
Comment (hyperupcall) [inline bin/git-rename-file:78]: Small nit, but it would be good to redirect all errors to standard error. I believe that's consistent with the rest of the code;

**Review by hyperupcall — CHANGES_REQUESTED** [review]

**Comment by hyperupcall** [comment]
Thanks, this looks good - I had a few suggestions;

File: bin/git-rename-file
Comment (CodeByZach) [reply bin/git-rename-file:51]: > Is `| grep -q $base` any different from redirecting like `>/dev/null 2>&1`? I believe `find` should already filter out the path;

The method I used was tested in multiple different scenarios. Please ensure whatever method you suggest is still able to efficiently and reliably detect case-sensitivity regardless of the case-sensitivity of the filesystem.

**Comment by CodeByZach** [comment]
Apologies, just catching up here. What's left before this can be merged?

File: bin/git-rename-file
Comment (hyperupcall) [reply bin/git-rename-file:51]: Yeah I don't know, I think I see now what `-iname` does on case insensitive file systems, which explains the `grep`.

Comment (CodeByZach) [reply bin/git-rename-file:51]: > Yeah I don't know, I think I see now what `-iname` does on case insensitive file systems, which explains the `grep`.

So should I leave as-is?

**Comment by hyperupcall** [comment]
@CodeByZach A must is to have the lint check in CI pass (if it's a lot of issues, running `shfmt` locally should speed things up). While you do that it would be a good idea to address the code change recommendations

**Comment by CodeByZach** [comment]
> @CodeByZach Hey, a must is to have the lint check in CI pass (if it's a lot of issues, running `shfmt` locally should speed things up). While you do that it would be a good idea to address the code change recommendations

Seems easy enough.
image

File: bin/git-rename-file
Comment (hyperupcall) [reply bin/git-rename-file:51]: Oops, I missed this notification. Yeah, leave it as it is. My recommendation came from a misunderstanding

**Comment by hyperupcall** [comment]
@CodeByZach Thank you for the update. I'll be able to merge this once @spacewander's comments [here](https://github.com/tj/git-extras/pull/1149#discussion_r1677280688) and [here](https://github.com/tj/git-extras/pull/1149#discussion_r1678682251) are addressed

**Comment by CodeByZach** [comment]
> @CodeByZach Thank you for the update; I'll be able to merge this once @spacewander's comments [here](https://github.com/tj/git-extras/pull/1149#discussion_r1677280688) and [here](https://github.com/tj/git-extras/pull/1149#discussion_r1678682251) are addressed;

Should now be addressed.

**Review by spacewander — APPROVED** [review]

**Pull request merged**
